File: Capítulo_15/Complexidade de algoritmos/Compreendendo a notação big-O/O(n) - Linear/sequentialSearch.js
import { defaultEquals } from "../../../Models/util.js";

function sequentialSearch(array, value, equalsFn = defaultEquals) {
    let cost = 0;
    for (let i = 0; i < array.length; i++) {
        cost++;
        if (equalsFn(value, array[i])) {            
            console.log(
            `cost for sequentialSearch with input size ${array.length} is ${cost}`)
            return i;
        }
    }
    console.log(
        `cost for sequentialSearch with input size ${array.length} is ${cost}`
    );
    return -1;
}

let array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
sequentialSearch(array, 15)

File: Capítulo_15/Complexidade de algoritmos/Compreendendo a notação big-O/O(n) - Linear/Código em Python/SequentialSearch.py
# Function defaultEquals
def default_equals(a, b):
    return a == b

def sequential_search(array, value, equals_fn = default_equals):
    cost = 0
    for i in range(len(array)):
        cost += 1
        if equals_fn(value, array[i]):
            print(f"cost for sequential_search with input size {len(array)} is {cost}")
            return i
    
    print(f"cost for sequential_search with input size {len(array)} is {cost}")
    return -1

array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
sequential_search(array, 21)
